pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConfig.java

```diff
@@ -126,6 +126,7 @@ public StreamConfig(String tableNameWithType, Map<String, String> streamConfigMa
     String consumerFactoryClassKey =
         StreamConfigProperties.constructStreamProperty(_type, StreamConfigProperties.STREAM_CONSUMER_FACTORY_CLASS);
     // For backward compatibility, default consumer factory is for Kafka.
+    // TODO: remove this default and make it mandatory to have a factory class
```

Comment:
I don't follow this TODO. We want to default to low level Kafka

Reply:
I thought we shouldn't have a default stream system. more strict config checks. Can remove the comment, if you disagree.

Reply:
IMO it is okay to use Kafka by default, as long as we document it